packages/flutter_tools/lib/src/android/gradle_log_processor.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'gradle_errors.dart';
+
+/// Process log output from gradle, removing irrelevant output and capturing
+/// exception information.
+class GradleLogProcessor {
+  GradleLogProcessor(this.localGradleErrors, this.verbose);
+
+  final List<GradleHandledError> localGradleErrors;
+  final bool verbose;
+
+  GradleHandledError detectedGradleError;
+  String detectedGradleErrorLine;
+  bool atFailureFooter = false;
+
+  String consumeLog(String line) {
+    // All gradle failures lead to a fairly long footer which contains mostly
+    // irrelevant information for a flutter build, along with misleading advice to
+    // run with --stacktrace (which does not exist for the flutter CLI). remove this.
+    if (!verbose && (line.startsWith('FAILURE: Build failed with an exception.') || atFailureFooter)) {
```

Comment:
Is there any way split on output that we control instead of a line from gradle? Like could the tool emit a known line from the recursive call?

Reply:
Can you elaborate a bit on what that would be used for?

Reply:
Sure, just taking a possibly naive look at the example 'before' output in the PR description, we have:
```
[lines emitted by the tool or the Dart compiler]
FAILURE: Build failed with an exception. [First line emitted by gradle?]
[more lines emitted by gradle]
[more lines emitted by the tool]
```
Instead of detecting gradle output using the first line emitted by gradle, which we don't control, the idea would be to detect gradle output using the last line of `[lines emitted by the tool or the Dart compiler]`, which we do control, adding a new sentinel line there if needed.

Reply:
We can't hide all Gradle output though. For example, a compilation error in the users Java code:

```
If you are deploying the app to the Play Store, it's recommended to use app bundles or split the APK to reduce the APK size.
    To generate an app bundle, run:
        flutter build appbundle --target-platform android-arm,android-arm64,android-x64
        Learn more: https://developer.android.com/guide/app-bundle
    To split the APKs per ABI, run:
        flutter build apk --target-platform android-arm,android-arm64,android-x64 --split-per-abi
        Learn more: https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
/Users/jonahwilliams/Documents/flutter/dev/integration_tests/flutter_gallery/android/app/src/main/java/io/flutter/demo/gallery/MainActivity.java:11: error: '{' expected
public class MainActivity extends FlutterActivity                       
                                                 ^                      
1 error                                                                 
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:compileReleaseJavaWithJavac'.           
> Compilation failed; see the compiler error output for details.        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 55s                  
```

We still want to chop off the footer since it is wrong, but we need to show the output from the javac task

Reply:
Ah, okay. Got it. Thanks.